### ISSUE TITLE
In game movable primitives?

### ISSUE THREAD

**Issue by Scottacus64** [vpinball/vpinball#630]
There are a few times when having primitives that can be moved during game play would be awesome.  A few examples are the objects on the playfield in my "Nudgy" table, the spinner on Bord's "Road Race" table and the rotating pop bumpers in Williams "Nags" table.  Would there be any way of adding this as a feature to vpx?  The ability to rotate and change the x/y coordinates of an abject.

**Comment by derwin12** [comment]
This is already possible .. I have a moveable Munster Herman head for example.

**Comment by Scottacus64** [comment]
Yes you can move the visual aspect of a prim but can you can move the physical collision aspect of the prim?

**Comment by wiesshund** [comment]
careful with collidable prims
The generally have a lot more faces and points of contact
(the visual prim you actually want to look at)

That can play hell with frame rate.

I am pretty sure they can collide though.
You can find out easy.

You know how to make a slingshot with a little prim kicker foot right?
ok make a ball pocket for the ball to roll in

Take a little kicker foot or something
put it in the pocket and set it do holding a key moves it
see if it changes where the ball stops
or make a simple post prim and raise and lower it, see if it blocks the ball raise and doesnt block when it lowers below the playfield

just watch how many polys you are allowing collision with.
I once accidentally flipped my plastic pegs to collidable
the result was horrid when the ball ran into them

**Comment by Scottacus64** [comment]
I once did the same thing with making pegs have collide properties and the game does get pretty funky.  You make good points about the number of faces and points of contact.  I'll have to try out what you say but I don't think that you can move the physical aspects of a primitive.
 
 What I'm looking for is straight lines or circles that can move and collide.  That would take care of moving bumpers and spinning "bars" that you find on certain older pins.

**Comment by wiesshund** [comment]
pretty sure you can considering if i put in prims for flippers
and make them collidable
and then dont put them ON the actual flipper
i get a dumb flipper in the middle of the playfield whacking the ball god knows where
